hushline/settings.py

```diff
@@ -224,57 +273,65 @@ def index() -> str | Response:
-            user.extra_field_value4 = profile_form.extra_field_value4.data.strip()
-            db.session.commit()
-            flash("👍 Bio updated successfully.")
-            return redirect(url_for("settings.index"))
-
-        if request.method == "POST" and (
-            directory_visibility_form.validate_on_submit()
-            and "update_directory_visibility" in request.form
-        ):
-            user.show_in_directory = directory_visibility_form.show_in_directory.data
-            db.session.commit()
-            flash("👍 Directory visibility updated successfully.")
-            return redirect(url_for("settings.index"))
+        # Handle form submissions
+        if request.method == "POST":
+            # Update bio and custom fields
+            if "update_bio" in request.form and profile_form.validate_on_submit():
+                user.bio = profile_form.bio.data
+
+                async def perform_verification() -> None:
+                    async with aiohttp.ClientSession() as session:
+                        tasks = []
+                        for i in range(1, 5):
+                            label_field = getattr(profile_form, f"extra_field_label{i}", "")
+                            value_field = getattr(profile_form, f"extra_field_value{i}", "")
+
+                            label = (
+                                label_field.data if hasattr(label_field, "data") else label_field
+                            )
+                            setattr(user, f"extra_field_label{i}", label)
+
+                            value = (
+                                value_field.data if hasattr(value_field, "data") else value_field
+                            )
+                            setattr(user, f"extra_field_value{i}", value)
+
+                            # If the value is empty, reset the verification status
+                            if not value.strip():
+                                setattr(user, f"extra_field_verified{i}", False)
+                                continue
+
+                            # Verify the URL only if it starts with "https://"
+                            url_to_verify = value.strip()
+                            if url_to_verify.startswith("https://"):
+                                profile_url = (
+                                    f"https://tips.hushline.app/to/{user.primary_username}"
+                                )
+                                # Schedule async task for verification
+                                task = verify_url(session, user, i, url_to_verify, profile_url)
+                                tasks.append(task)
+
+                        # Run all the tasks concurrently
+                        if tasks:  # Only gather if there are tasks to run
+                            await asyncio.gather(*tasks)
+
+                # Run the async verification function
+                asyncio.run(perform_verification())
```

Comment:
I don't think this needs to be an async function if we change the signature of the route function to be async because flask should invoke the route handler, see the snippet at the top of https://flask.palletsprojects.com/en/3.0.x/async-await/